### PR TITLE
Add Dart example.

### PR DESCRIPTION
**Pull request by szakarias**

### CODE REVIEW

File: examples/add_person.dart [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  AddressBook addressBook;
+  if (!file.existsSync()) {
+    print('File not found. Creating new file.');
+    addressBook =  AddressBook();
+  } else {
+    addressBook = AddressBook.fromBuffer(file.readAsBytesSync());
+  }
+  addressBook.people.add(promtForAddress());
+  file.writeAsBytes(addressBook.writeToBuffer());
+}
```
Comment (sigurdm) [inline examples/add_person.dart]: Add trailing newline / format with dartfmt

File: examples/README.md [hunk 2]
```diff
@@ -122,3 +122,22 @@ is created if it does not exist.  To view the data, run:
 Observe that the C++, Python, and Java examples in this directory run in a
```
Comment (sigurdm) [inline examples/README.md]: Add dart to this sentence

File: examples/README.md [hunk 3]
```diff
@@ -122,3 +122,22 @@ is created if it does not exist.  To view the data, run:
...
+First, follow the instructions in [../README.md](../README.md) to install the Protocol Buffer Compiler (protoc).
+
+Then, install the Dart Protocol Buffer plugin as described [here](https://github.com/dart-lang/dart-protoc-plugin#how-to-build-and-use).
+Note, the executable `bin/protoc-gen-dart` must be in your `PATH` for `protoc` to find it.
+
+Build the Dart samples in this directory with `make dart`.
+
+To run the examples:
+
+    $ dart add_person.dart addessbook.data
```
Comment (sigurdm) [inline examples/README.md]: Perhaps use
````
```sh
$ dart add_person.dart addessbook.data
$ dart list_people.dart addressbook.data
```
````

File: examples/README.md [hunk 4]
```diff
@@ -134,9 +134,10 @@ Build the Dart samples in this directory with `make dart`.
 
 To run the examples:
 
+```sh
     $ dart add_person.dart addessbook.data
```
Comment (sigurdm) [inline examples/README.md]: Nit: you don't need indentation when you use triple quotes.

**Comment by anandolee** [comment]
Protobuf team does not maitain dart, it is a little wired to add dart example here. I noticed there's a third party of dart:
https://github.com/dart-lang/dart-protobuf (runtime) https://github.com/dart-lang/dart-protoc-plugin (code generator)

Is the example related to the third party version? If so I'd prefer to add the example to the third party web

**Comment by anandolee** [comment]
Emailed with Michael Thomsen, dart is a google-funded language so it make sense to add example here same with go languages.

I am not a native English speaker. @gerben-s and @haberman are you able to take a look at the documents?

**Comment by mit-mit** [comment]
We have other documentation that is blocked on this getting in first; @gerben-s and @haberman can you please take a look?

**Pull request merged**